src/main/java/seedu/address/model/person/Address.java

```diff
@@ -9,13 +9,15 @@
  */
 public class Address {
 
-    public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
+    private static final int MAXIMUM_LENGTH = 400;
+    public static final String MESSAGE_CONSTRAINTS = "Addresses should not be blank, "
+            + "and it has no more than" + MAXIMUM_LENGTH + "characters(excluding blank space at front)";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String VALIDATION_REGEX = "[^\\s].{0," + MAXIMUM_LENGTH + "}";
```

Comment:
I did some light testing and the `VALIDATION_REGEX` allows `401` characters, which is more than the `MAXIMUM_LENGTH` allowed.